pkg/apis/pipeline/v1/matrix_types.go

```diff
@@ -96,11 +97,47 @@ func createCombination(name string, value string, combination Params) Params {
 
 // CountCombinations returns the count of combinations of Parameters generated from the Matrix in PipelineTask.
 func (m *Matrix) CountCombinations() int {
+	// Explicit combinations in the Matrix
+	if m.hasInclude() && !m.hasParams() {
+		return len(m.Include)
+	}
+
+	// Iterate over matrix.params and compute count of all generated combinations
+	count := countMatrixParamsCombinations(m.Params)
+
+	if m.hasInclude() {
+		paramMap := m.Params.extractParamMapArrVals()
```

Comment:
`paramMap` can also be part of the function `countNewCombinationsFromInclude` instead of defining it here. And once `paramMap` is part of the function `countNewCombinationsFromInclude`, if `m.hasInclude` is redundant since we are iterating over an array `m.Include`.

Reply:
Ah I see. I refactored all of these to member functions and will keep this in mind for the implementation as well. Thanks!